graphite_api/app.py

```diff
@@ -477,9 +487,28 @@ def render():
     return response
 
 
-def evaluateTarget(requestContext, target):
+def pathsFromTarget(target):
+    paths = []
+    tokens = grammar.parseString(target)
+    pathsFromTokens(tokens, paths)
+    return paths
+
+
+def pathsFromTokens(tokens, paths):
+    if tokens.expression:
+        pathsFromTokens(tokens.expression, paths)
+    elif tokens.pathExpression:
+        paths.append(tokens.pathExpression)
+    elif tokens.call:
+        for arg in tokens.call.args:
+            pathsFromTokens(arg, paths)
+        for kwarg in tokens.call.kwargs:
+            pathsFromTokens(kwarg.args[0], paths)
```

Comment:
Could you rewrite that function to return a list of paths instead of mutating an existing list? Something like

```python
def pathsFromTokens(tokens):
    if tokens.expression:
        for path in pathsFromTokens(tokens.expression):
            yield path
    elif tokens.pathExpression:
        yield tokens.pathExpression
    elif tokens.call:
        for arg in token.call.args:
            for path in pathsFromTokens(arg):
                yield path
        for kwarg in tokens.call.kwargs:
            for path in pathsFromTokens(kwarg.args[0]):
                yield path
```

Then `pathsFromTarget` would become:

```python
def pathsFromTarget(target):
    tokens = grammar.parseString(target)
    return list(pathsFromTokens(tokens))
```

(the `list()` call is just here to force evaluation of `pathsFromTokens` since it's a generator).